release.sh

```diff
@@ -98,6 +103,29 @@ commit_belongs_to_release_branch() {
     return 1
 }
 
+new_snapshot () {
+    local sha latest latest_prefix prefix new_line tmp
+    sha=$(git rev-parse HEAD)
+    if ! commit_belongs_to_release_branch $sha; then
+        echo "WARNING: Commit does not belong to a release branch."
+    fi
+    latest=$(head -1 LATEST | awk '{print $2}')
+    latest_prefix=$(echo $latest | grep -o -P "$STABLE_REGEX" | head -1)
+    if is_stable $latest; then
+        prefix=$(echo $latest_prefix | jq -Rr '. | split(".") | [.[0], (.[1] | tonumber + 1 | tostring), "0"] | join(".")')
+    else
+        prefix=$latest_prefix
+    fi
+    new_line=$(make_snapshot $sha $prefix)
+    tmp=$(mktemp)
+    cp LATEST $tmp
+    if is_stable $latest; then
```

Comment:
I don’t think this works given the overlap we have atm. Not a huge deal given that this happens once every month but maybe removing it completely is a better option then.

Reply:
I'm not sure I follow. In which case would this not work? If the stable release PR was opened on Tuesday evening and merged on Wednesday afternoon? Or are you concerned we may "overshoot" and end up with a stable release that is "before" (in commit order) the latest snapshot for that prefix? I'm not entirely sure that's an issue, but worth thinking about a bit more I suppose. 🤔

Maybe we'd be better served by two files, `SNAPSHOTS` and `RELEASES`... 🤔

Reply:
We need a snapshot for 1.4 before 1.3 is released. When we release 1.3 and this check could trigger we already have a snapshot for 1.4 so this doesn’t do anything.

Reply:
Oh, so what you mean is not that this is an issue, just that if everything else works as planned we should never actually have a LATEST file where the first line is a stable release?

If that's what you meant, I still think it's better that the script does "the right thing" should that ever happen.

Reply:
Not quite, I think this actually gives you a false security that it will bump in the correct circumstances and stop people from thinking about whether we already branched off for a release and they need to bump or not.

Reply:
Ah, right. People are not supposed to be involved here. As you may have noticed I have not updated the release instructions.

This PR is actually missing its most important file. Sorry about that; added now. (`ci/cron/wednesday.yml`)

Does that make it better?

Reply:
For future readers, we took a chunk of that discussion offline. The end result is I've added a comment in the `release.sh` script, as well as a body to the automatically-generated PR, which hopefully together address the concern raised by @cocreature.